### PR TITLE
[RFC] vim-patch:7.4.2160

### PR DESCRIPTION
**Pull request by Shougo**
Problem:    setmatches() mixes up values. (Nikolai Pavlov)
Solution:   Save the string instead of reusing a shared buffer.

https://github.com/vim/vim/commit/7dc5e2e486fe0287601968e535902a41a39f65bb

### CODE REVIEW

**Comment by justinmk** [comment]
This is missing a comment added to `dict_find()`.

**Comment by Shougo** [comment]
Fixed.

File: src/nvim/testdir/test_expr.vim [hunk 1]
```diff
@@ -96,4 +96,12 @@ endfunc
 func Test_special_char()
   " The failure is only visible using valgrind.
   call assert_fails('echo "\<C-">')
+
```
Comment (brcolow) [inline src/nvim/testdir/test_expr.vim]: Is an endfunc missing here?

**Comment by Shougo** [comment]
Thank you.

**Pull request merged**